src/storage/src/source/source_reader_pipeline.rs

```diff
@@ -436,33 +436,16 @@ where
 
             async move {
                 // Setup time!
-                let mut source_upper = {
-                    let upper_ts = resume_upper.as_option().copied().unwrap();
-                    let as_of = Antichain::from_elem(upper_ts.saturating_sub(1));
-                    // This is a TEMPORARY instance of a `ReclockOperator` used to calculate
-                    // a OFFSET `source_upper` for the SourceReader. It cannot be
-                    // shared with the `remap_operator` because its initialization
-                    // is async, which is not acceptable within dataflow creation.
-                    // It cannot be a `ReclockFollower`, which is empty on
-                    // initialization.
-                    let timestamper = match ReclockOperator::new(
-                        Arc::clone(&persist_clients),
-                        storage_metadata.clone(),
-                        now.clone(),
-                        timestamp_interval.clone(),
-                        as_of,
-                    )
-                    .await
-                    {
-                        Ok(t) => t,
-                        Err(e) => {
-                            panic!("Failed to create source {} timestamper: {:#}", name, e);
-                        }
-                    };
-                    timestamper
-                        .source_upper_at_frontier(resume_upper.borrow())
-                        .expect("source_upper_at_frontier to be used correctly")
-                };
+
+                // required to build the initial source_upper and to ensure the offset committer
+                // operator correctly.
```

Comment:
nit: something seems to be missing from that sentence. Maybe a "can work correctly"?